machine-learning/ml_tips/README.md

```diff
@@ -0,0 +1,74 @@
+# ML tips workflow
+
+This workflow aims to explain common ML techniques using Hivemall and Treasure Workflow.
+
+Using [titanic data](https://github.com/amueller/scipy-2017-sklearn/blob/master/notebooks/datasets/titanic3.csv), this workflow explains the following basic techniques.
+
+- Random/stratified splitting for train and test data
+- Feature normalization
+- Missing value imputation
+
+## Input
+
+This workflow assumes a table as follows:
+
+|pclass<br/>`int`|survived<br/>`int`|name<br/>`string`|sex<br/>`string`|age<br/>`int`|sibsp<br/>`int`|parch<br/>`int`|ticket<br/>`string`|fare<br/>`double`|cabin<br/>`string`|embarked<br/>`string`|boat<br/>`string`|body<br/>`int`|home.dest<br/>`string`|
+|---------|------|---------|--------|-------|-------|-------|--------|-------|-------|-----------|--------|---------|-------|
+|1|1|"Allen, Miss. Elisabeth Walton"|female|29|0|0|24160|211.3375|B5|S|2||"St Louis, MO"|
+|1|1|"Allison, Master. Hudson Trevor"|male|0.9167|1|2|113781|151.5500|C22 C26|S|11||"Montreal, PQ / Chesterville, ON"|
+|1|0|"Allison, Miss. Helen Loraine"|female|2|1|2|113781|151.5500|C22 C26|S|||"Montreal, PQ / Chesterville, ON"|
+|...|...|...|...|...|...|...|...|...|...|...|...|...|...|
+
+Here is a description of each column.
+
+    pclass          Passenger Class
+                    (1 = 1st; 2 = 2nd; 3 = 3rd)
+    survival        Survival
+                    (0 = No; 1 = Yes)
+    name            Name
+    sex             Sex
+    age             Age
+    sibsp           Number of Siblings/Spouses Aboard
+    parch           Number of Parents/Children Aboard
+    ticket          Ticket Number
+    fare            Passenger Fare
+    cabin           Cabin
+    embarked        Port of Embarkation
+                    (C = Cherbourg; Q = Queenstown; S = Southampton)
+    boat            Lifeboat
+    body            Body Identification Number
+    home.dest       Home/Destination
+
+In this workflow, you will predict the `survival` column as a binary classification problem.
```

Comment:
that's more precise. fixed 8cfd54b9d1ce0dca2694442bf80b521bab219a1f

Reply:
```
predict the provability of survival using logistic regression, taking a binary label survival as a response variable.
```

would be more precise that users can know behind the scenes.

Reply:
I didn't intend to restrict this workflow to only for logistic regression since I'm trying to add RandomForest example, currently, I couldn't figure out to add it though. 

How about fixing like this? 

```suggestion
In this workflow, you will predict the probability of `survival`, taking a binary label survival as a response variable.
```

Reply:
For RandomForest, it's hard to share the preprocessing process because most of feature engineering is not required in RF.

Reply:
Out of curiosity, as far as I know, Random Forest doesn't need to normalize values or to impute missing values, but it should work with the same preprocessing as Logistic Regression. Is there any downside or Hivemall specific restriction on it?

Reply:
I confirmed Random Forest with standardization, scaling with zscore, doesn't work appropreately. It'd be helpful if there is some assumption which I missed.

Reply:
Fixed the description again, without mentioning LogisticRegression though. 0915eb9